zypper.py

```diff
@@ -0,0 +1,274 @@
+    gauge = Gauge(prefix,
+                  description,
+                  namespace=NAMESPACE,
+                  registry=REGISTRY)
+    filters = filters or {}
+    if len(data) == 0:
+        gauge.set(0)
+    else:
+        for package in data:
+            check = all(package.get(k) == v for k, v in filters.items())
+            if check:
+                gauge.inc()
+
+
+def print_reboot_required():
+    needs_restarting_path = '/usr/bin/needs-restarting'
+    is_path_ok = os.path.isfile(needs_restarting_path) and os.access(needs_restarting_path, os.X_OK)
+
+    if is_path_ok:
+        prefix = "node_reboot_required"
+        description = (
+            "Node require reboot to activate installed updates or patches. "
+            "(0 = not needed, 1 = needed)"
+        )
+        info = Info(prefix, description)
+        result = subprocess.run(
+            [needs_restarting_path, '-r'],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=False)
+        if result.returncode == 0:
+            info.info({"node_reboot_required": "0"})
+        else:
+            info.info({"node_reboot_required": "1"})
+
+
+def print_zypper_version():
+    result = subprocess.run(
+        ['/usr/bin/zypper', '-V'],
+        stdout=subprocess.PIPE,
+        check=False).stdout.decode('utf-8')
+    info = Info("zypper_version", "zypper installed package version")
+
+    info.info({"zypper_version": result.split()[1]})
+
+
+def __extract_data(raw, fields):
+    raw_lines = raw.splitlines()[2:]
+    extracted_data = []
+
+    for line in raw_lines:
+        parts = [part.strip() for part in line.split('|')]
+        if len(parts) >= max(fields.values()) + 1:
+            extracted_data.append({
+                field: parts[index] for field, index in fields.items()
+            })
+
+    return extracted_data
+
+
+def stdout_zypper_command(command):
+    result = subprocess.run(
+        command,
+        stdout=subprocess.PIPE,
+        check=False
+    )
+
+    if result.returncode != 0:
+        raise RuntimeError(f"zypper returned exit code {result.returncode}: {result.stderr}")
+
+    return result.stdout.decode('utf-8')
+
+
+def extract_lu_data(raw: str):
+    fields = {
+        "Repository": 1,
+        "Name": 2,
+        "Current Version": 3,
+        "Available Version": 4,
+        "Arch": 5
+    }
+
+    return __extract_data(raw, fields)
+
+
+def extract_lp_data(raw: str):
+    fields = {
+        "Repository": 0,
+        "Name": 1,
+        "Category": 2,
+        "Severity": 3,
+        "Interactive": 4,
+        "Status": 5
+    }
+
+    return __extract_data(raw, fields)
+
+
+def extract_orphaned_data(raw: str):
+    fields = {
+        "Name": 3,
+        "Version": 4
+    }
+
+    return __extract_data(raw, fields)
+
+
+def __parse_arguments(argv):
+    parser = argparse.ArgumentParser()
+
+    parser.add_mutually_exclusive_group(required=False)
```

Comment:
This looks incorrect. I think what you meant to do is
```python
group = parser.add_mutually_exclusive_group(required=False)
group.add_argument("-m", ...)
group.add_argument("-l", ...)
```
However, this combination of "more / less" in a mutually exclusive group, _and_ with a default that sets the `--more` option true (thus overriding the fact you set the group `required=False`), is **really** unintuitive from a user's perspective.

You have essentially just reinvented a boolean option, which by definition can only be one of two values, i.e. mutually exclusive.

Please re-work this to something that is a bit clearer.
```